flashinfer/fused_moe/core.py

```diff
@@ -868,12 +918,268 @@ def trtllm_gen_fused_moe_sm100_module() -> JitSpec:
     )
 
 
+def deduce_trtllm_gen_tensor_dtype(
+    x: torch.Tensor, scale: Optional[torch.Tensor]
+) -> DtypeTrtllmGen:
+    hidden_size = x.shape[-1]
+    if x.dtype == torch.uint8:
+        hidden_size *= 2
+    if x.dtype == torch.bfloat16:
+        dtype = DtypeTrtllmGen.Bfloat16
+    elif x.dtype == torch.float8_e4m3fn:
+        dtype = DtypeTrtllmGen.E4m3 if scale is None else DtypeTrtllmGen.MxE4m3
+    elif x.dtype == torch.uint8:
```

Comment:
Just a minor note: we should also take care of `torch.float4_e2m1x2` for torch 2.8+

Reply:
Not yet, we need to prepare for it when frameworks are all upgrading to torch 2.8.

It could be done in later PRs.